File: 1833.雪糕的最大数量.py
#
# @lc app=leetcode.cn id=1833 lang=python3
#
# [1833] 雪糕的最大数量
#
from typing import List
# @lc code=start
class Solution:
    def maxIceCream(self, costs: List[int], coins: int) -> int:
        costs.sort()
        res = 0
        for i in range(len(costs)):
            if coins >= costs[i]:
                coins -= costs[i]
                res += 1
            else:
                break
        return res
# @lc code=end

File: 1833.雪糕的最大数量.go
/*
 * @lc app=leetcode.cn id=1833 lang=golang
 *
 * [1833] 雪糕的最大数量
 */

// @lc code=start
func maxIceCream(costs []int, coins int) int {
	sort.Ints(costs)
	res := 0
	for _, v := range costs {
		if coins >= v {
			coins -= v
			res++
		} else {
			break
		}
	}
	return res

}

// @lc code=end

